academic-services/knowledge-exploration-service/reference-makes-command-line-tool.md

```diff
@@ -171,37 +196,287 @@ The input entities file Url prefix.
+
+`--Count`
+
+The number of top interpretations to be included in the result set.
+
+`--SelectInterpretationEntities`
+
+A list of entity attributes, seperated by ','. Use '*' for all entity attributes. Each interpretation in the result set can include the top matching entities used for generating the interpretation. SelectInterpretationEntities specifies which attributes of the top matching entities should be included in the result set.
+
+`--TakeInterpretionEntities`
+
+The number of top matching entities to be included for each interpretation. Each interpretation in the result set can include the top matching entities used for generating the interpretation. TakeInterpretionEntities specifies how many top matching entnties should be included in the result set
+
+`--Timeout`
+
+Maximum amount of time in milliseconds allowed for command to complete before aborting the command. The interpret command will return the top interptations found in the allowed timedout.
+
+## Evaluate Command
+
+Evaluates a KES query expression and returns the top matching entities in the index(es).
+
+```cmd
+kesm Evaluate --KesQueryExpression
+              --IndexFilePaths
+              [--Attributes]
+              [--Skip]
+              [--Take]
+              [--OrderBy]
+              [--OrderByDescending]
+              [--Timeout]
+```
+
+### Evaluate Required Parameters
+
+`--KesQueryExpression`
+
+The KES query expression that specifies entities in the index(es).
+
+`--IndexFilePaths`
+
+The file path expression for specifying which index file(s) to use. Use wild card to specify multiple indexes e.g. './index.*.kes'
+
+### Evaluate Optional Parameters
+
+`--Attributes`
+
+A list of entity attributes to be included in the result set, seperated by ','. Use '*' for all attributes.
+
+`--Offset`
+
+The number of top entities to be skipped/excluded in the result set.
+
+`--Count`
+
+The number of top interpretations to be included in the result set.
+
+`--OrderBy`
+
+Name of the entity attribute to use for sorting/ordering the entities in the result set. By default, entities are sorted by descending entity weight (static rank).
+
+`--OrderByDescending`
+
+The direction for sorting entities. By default, entities are sorted by descending entity weight (static rank).
+
+`--Timeout`
+
+Maximum amount of time in milliseconds allowed for the command to complete before aborting the command. Use 0 to disable timeout.
+
+## Histogram Command
+
+Calculates distinct/total entity attribute counts and top attribute values for entities specified by a KES query expression.
+
+```cmd
+kesm Histogram --KesQueryExpression
+               --IndexFilePaths
+               [--Attributes]
+               [--Skip]
+               [--Take]
+               [--OrderBy]
+               [--OrderByDescending]
+               [--Timeout]
+```
+
+### Histogram Required Parameters
+
+`--KesQuery`
+
+The KES query expression that specifies entities in the index(es).
```

Comment:
The [query expression](concepts-query-expressions.md) to use for selecting entities from the index(es).